src/components/tag/tag.classnames.ts

```diff
@@ -1,80 +1,365 @@
+  LabelPairedCircleInfoRegularIcon,
   QuillSvgProps,
 } from '@deriv/quill-icons'
+import { TPreset, TSize } from '.'
 
-type TypeTagColor = {
-  [key in TagType]: {
-    [key in TagVariant]: string
+type TTagIcon = {
+  [key in Exclude<TPreset, 'custom'>]: {
+    [key in 'true' | 'false']: React.ForwardRefExoticComponent<
+      Omit<QuillSvgProps, 'ref'>
+    >
   }
 }
 
-export const TagColors: TypeTagColor = {
+export const TagIcons: TTagIcon = {
   error: {
-    fill: 'bg-solid-red-100 text-solid-red-800',
-    outline: 'border-75 border-solid-red-700 text-solid-red-800',
+    true: LabelPairedTriangleExclamationBoldIcon,
+    false: LabelPairedTriangleExclamationRegularIcon,
   },
   warning: {
-    fill: 'bg-solid-orange-100 text-solid-orange-800',
-    outline: 'border-75 border-solid-orange-700 text-solid-orange-800',
+    true: LabelPairedCircleExclamationBoldIcon,
+    false: LabelPairedCircleExclamationRegularIcon,
+  },
+  success: {
+    true: LabelPairedCircleCheckBoldIcon,
+    false: LabelPairedCircleCheckRegularIcon,
   },
   info: {
-    fill: 'bg-solid-blue-100 text-solid-blue-800',
-    outline: 'border-75 border-solid-blue-700 text-solid-blue-800',
+    true: LabelPairedCircleInfoBoldIcon,
+    false: LabelPairedCircleInfoRegularIcon,
   },
-  success: {
-    fill: 'bg-solid-green-100 text-solid-green-800',
-    outline: 'border-75 border-solid-green-700 text-solid-green-800',
+}
+
+export const TagIconSizes: Record<TSize, { width: number; height: number }> = {
+  xs: {
+    width: 11,
+    height: 18,
+  },
+  sm: {
+    width: 11,
+    height: 18,
+  },
+  md: {
+    width: 13,
+    height: 22,
+  },
+  lg: {
+    width: 14,
+    height: 24,
   },
 }
 
-export const TagSizes: Record<TagSize, string> = {
-  xs: 'px-200 py-100 text-50 gap-x-200',
-  sm: 'px-400 py-[3px] text-50 gap-x-400',
-  md: 'px-600 py-[5px] text-75 gap-x-600',
-  lg: 'px-800 py-600 text-100 gap-x-800',
+export const TagIconColors = {
+  error: 'fill-status-danger',
+  warning: 'fill-status-warning',
+  success: 'fill-status-success',
+  info: 'fill-status-info',
 }
 
-export const TagIcons: Record<
-  TagType,
-  React.ForwardRefExoticComponent<Omit<QuillSvgProps, 'ref'>>
-> = {
-  error: LabelPairedTriangleExclamationBoldIcon,
-  warning: LabelPairedCircleExclamationBoldIcon,
-  success: LabelPairedCircleCheckBoldIcon,
-  info: LabelPairedCircleInfoBoldIcon,
+export const TagCustomIconColors = {
+  red: 'fill-solid-red-1200',
+  orange: 'fill-solid-orange-1200',
+  yellow: 'fill-solid-yellow-1200',
+  mustard: 'fill-solid-mustard-1200',
+  olive: 'fill-solid-olive-1200',
+  green: 'fill-solid-green-1200',
+  tiffany: 'fill-solid-tiffany-1200',
+  teal: 'fill-solid-teal-1200',
+  seawater: 'fill-solid-seawater-1200',
+  blue: 'fill-solid-blue-1200',
+  sapphire: 'fill-solid-sapphire-1200',
+  blueberry: 'fill-solid-blueberry-1200',
+  grape: 'fill-solid-grape-1200',
+  magenta: 'fill-solid-magenta-1200',
+  slate: 'fill-solid-slate-1200',
```

Comment:
Is it possible to move these into CVA?

Reply:
This object is to target the Icon within the component. I'll try to move it into the cva if it's possible. 👍🏼